src/java.base/share/classes/sun/security/provider/certpath/RevocationChecker.java

```diff
@@ -740,9 +746,44 @@ private void checkOCSP(X509Certificate cert,
                         null, -1);
                 }
 
+                List<Extension> tmpExtensions = Collections.<Extension>emptyList();
+                if (rp.ocspNonce) {
+                    if (nonce == null) {
+                        try {
+                            // create the 16-byte nonce by default
+                            Extension nonceExt = new OCSPNonceExtension(DEFAULT_NONCE_BYTES);
+
+                            if (ocspExtensions.size() > 0) {
+                                tmpExtensions = new ArrayList<Extension>(ocspExtensions);
+                            } else {
+                                tmpExtensions = new ArrayList<Extension>();
+                            }
+
+                            tmpExtensions.add(nonceExt);
```

Comment:
If you only need the nonce, you could use List.of and save a little bit of memory, ex:

                            if (ocspExtensions.size() > 0) {
                                tmpExtensions = new ArrayList<Extension>(ocspExtensions);
                                tmpExtensions.add(nonceExt);
                            } else {
                                tmpExtensions = List.of(nonceExt);
                            }

Reply:
Thanks for the review. Updated as suggested.